### PR TITLE
[Messenger] Allow sender tag name omission

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes  
| Fixed tickets | na
| License       | MIT
| Doc PR        | na

Tag attributes are not mandatory.

### CODE REVIEW

**Comment by sroze** [comment]
Could you add a test, please?

**Review by sroze — APPROVED** [review]

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 1]
```diff
@@ -157,7 +157,11 @@ private function registerReceivers(ContainerBuilder $container)
         $receiverMapping = array();
         foreach ($container->findTaggedServiceIds('messenger.receiver') as $id => $tags) {
             foreach ($tags as $tag) {
-                $receiverMapping[$tag['name'] ?? $id] = new Reference($id);
+                $receiverMapping[$id] = new Reference($id);
+
+                if ($tag['name'] ?? false) {
```
Comment (fabpot) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: I would use `isset($tag['name'])` instead.

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 2]
```diff
@@ -171,7 +175,7 @@ private function registerSenders(ContainerBuilder $container)
             foreach ($tags as $tag) {
                 $senderLocatorMapping[$id] = new Reference($id);
 
-                if ($tag['name']) {
+                if ($tag['name'] ?? false) {
```
Comment (fabpot) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: same here

**Review by fabpot — APPROVED** [review]
only minor changes.

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: it wouldn't mean the same, as falsy value should be excluded.
Could be `!empty($tag['name'])` if you prefer

Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: (but maybe we don't care? :) )

Comment (fabpot) [reply src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: Exactly, the current version is wrong to me, `isset()` is semantically better here.

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 3]
```diff
@@ -157,7 +157,11 @@ private function registerReceivers(ContainerBuilder $container)
         $receiverMapping = array();
         foreach ($container->findTaggedServiceIds('messenger.receiver') as $id => $tags) {
             foreach ($tags as $tag) {
-                $receiverMapping[$tag['name'] ?? $id] = new Reference($id);
+                $receiverMapping[$id] = new Reference($id);
+
+                if (isset($tag['name'])) {
+                    $receiverMapping[$tag['name']] = new Reference($id);
```
Comment (chalasr) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: `=  $receiverMapping[$id];`

**Comment by sroze** [comment]
Good catch, thanks @soyuka.

**Comment by sroze** [comment]
And well done for your first contribution 👏

**Pull request merged**